controllers/virtualmachine/volumebatch/volumebatch_controller.go

```diff
@@ -646,38 +664,28 @@ func (r *Reconciler) updateVMVolumeStatus(
 		if volStatus, ok := existingAttachVolStatus[vol.Name]; ok &&
 			vol.PersistentVolumeClaim.ClaimName == volStatus.PersistentVolumeClaim.ClaimName {
 
-			vmVolStatus = vmopv1.VirtualMachineVolumeStatus{
-				Name:     volStatus.Name,
-				Type:     vmopv1.VolumeTypeManaged,
-				Attached: volStatus.PersistentVolumeClaim.Attached,
-				DiskUUID: volStatus.PersistentVolumeClaim.DiskUUID,
-				Error:    pkgutil.SanitizeCNSErrorMessage(volStatus.PersistentVolumeClaim.Error),
-				Used:     existingVMManagedVolStatus[vol.Name].Used,
-				Crypto:   existingVMManagedVolStatus[vol.Name].Crypto,
-			}
+			vmVolStatus = attachmentStatusToVolumeStatus(volStatus.Name, volStatus)
+			vmVolStatus.Used = existingVMManagedVolStatus[vol.Name].Used
+			vmVolStatus.Crypto = existingVMManagedVolStatus[vol.Name].Crypto
 
 			// Add PVC capacity information
 			if err := r.updateVolumeStatusWithPVCInfo(
 				ctx,
 				volStatus.PersistentVolumeClaim.ClaimName,
 				&vmVolStatus); err != nil {
+
 				ctx.Logger.Error(err, "failed to get volume status limit")
 			}
 		}
 
 		volumeStatuses = append(volumeStatuses, vmVolStatus)
 	}
 
-	// Remove any managed volumes from the existing status.
-	ctx.VM.Status.Volumes = slices.DeleteFunc(ctx.VM.Status.Volumes,
-		func(e vmopv1.VirtualMachineVolumeStatus) bool {
-			return e.Type != vmopv1.VolumeTypeClassic
-		})
-
-	ctx.VM.Status.Volumes = append(ctx.VM.Status.Volumes, volumeStatuses...)
+	getVolumeStatusWithDetachingVolumeFromBatchAttachment(
+		existingAttachVolStatus, &volumeStatuses,
+	)
 
-	// Sort the volume statuses to ensure consistent ordering.
-	vmopv1.SortVirtualMachineVolumeStatuses(ctx.VM.Status.Volumes)
+	return volumeStatuses
 }
 
 func (r *Reconciler) updateVolumeStatusWithPVCInfo(
```

Comment:
Should we do a list of PVCs at the beginning to avoid calling "Get" multiple times?